File: Sem4/MPP/Lab/ProiectJava/ProjectNetworking/src/main/java/project/network/dto/DtoUtils.java
package project.network.dto;

import project.model.Bilet;
import project.model.Excursie;
import project.model.User;

import java.time.LocalDateTime;

public class DtoUtils {
	public static User getFromDto(UserDto userdto){
		int id = userdto.getId();
		String pass = userdto.getPassword();
		String username = userdto.getUsername();
		User usr = new User();
		usr.setUsername(username);
		usr.setPassword(pass);
		usr.setId(id);
		return usr;
	}

	public static UserDto getDto(User user){
		int id = user.getId();
		String username = user.getUsername();
		String password = user.getPassword();
		return new UserDto(id,username,password);
	}

	public static Bilet getFromDto(BiletDto biletdto){
		int id = biletdto.getId();
		int nrPers = biletdto.getNumarPersoane();
		String numeClient = biletdto.getNumeClient();
		int nrTel = biletdto.getNumarTelefonClient();
		int idEx = biletdto.getIdExcursie();
		Bilet bil = new Bilet(nrPers,numeClient,nrTel,idEx);
		bil.setId(id);
		return bil;
	}

	public static BiletDto getDto(Bilet bilet){
		int id = bilet.getId();
		int nrPers = bilet.getNumarPersoane();
		String numeClient = bilet.getNumeClient();
		int nrTel = bilet.getNumarTelefonClient();
		int idEx = bilet.getIdExcursie();
		return new BiletDto(nrPers,numeClient,nrTel,idEx,id);
	}
	public static Excursie getFromDto(ExcursieDto excursiedto){
		String obiectivVizitat = excursiedto.getObiectivVizitat();
		LocalDateTime oraPlecare = excursiedto.getOraPlecare();
		String firmaTransport = excursiedto.getFirmaTransport();
		Double pret = excursiedto.getPret();
		int locuriDisponibile = excursiedto.getLocuriDisponibile();
		int id = excursiedto.getId();
		Excursie exc = new Excursie(obiectivVizitat,oraPlecare,firmaTransport,pret,locuriDisponibile);
		exc.setId(id);
		return exc;
	}

	public static ExcursieDto getDto(Excursie excursie){
		String obiectivVizitat = excursie.getObiectivVizitat();
		LocalDateTime oraPlecare = excursie.getOraPlecare();
		String firmaTransport = excursie.getFirmaTransport();
		Double pret = excursie.getPret();
		int locuriDisponibile = excursie.getLocuriDisponibile();
		int id = excursie.getId();
		return new ExcursieDto(obiectivVizitat,oraPlecare,firmaTransport,pret,locuriDisponibile,id);
	}

	public static Excursie[] getFromDto(ExcursieDto[] excursiidto){
		Excursie[] excursii = new Excursie[excursiidto.length];
		for(int i =0;i< excursiidto.length;i++){
			excursii[i]=getFromDto(excursiidto[i]);
		}
		return excursii;
	}
	public static ExcursieDto[] getDto(Excursie[] excursii){
		ExcursieDto[] excursiiDto = new ExcursieDto[excursii.length];
		for (int i=0;i< excursii.length;i++){
			excursiiDto[i] = getDto(excursii[i]);
		}
		return excursiiDto;
	}



}

File: Sem4/MPP/Lab/ProiectCSharp/networking/DTOUtils.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using model;

namespace networking
{
    public class DTOUtils
    {
        public static UserDTO getDTO(User user)
        {
            return new UserDTO(user.username, user.password, user.id);
        }
        public static User fromDTO(UserDTO userdto) 
        {
            User usr = new User();
            usr.id = userdto.id;
            usr.password = userdto.password;
            usr.username = userdto.username;
            return usr;
        }

        public static BiletDTO getDTO(Bilet bilet)
        {
            return new BiletDTO(bilet);
        }

        public static Bilet fromDTO(BiletDTO biletdto)
        {
            int id = biletdto.id;
            String numeClient = biletdto.numeClient;
            int numarTelefonClient = biletdto.numarTelefonClient;
            int idExcursie = biletdto.idExcursie;
            int numarPersoane = biletdto.numarPersoane;
            Bilet bilet = new Bilet(numarPersoane,numeClient,numarTelefonClient,idExcursie);
            bilet.setId(id);
            return bilet;
        }
        public static ExcursieDTO getDTO(Excursie excursie)
        {
            return new ExcursieDTO(excursie);
        }
        public static Excursie fromDTO(ExcursieDTO excursiedto)
        {
            int id = excursiedto.id;
            String obiectiv = excursiedto.obiectivVizitat;
            DateTime ora = DateTime.Parse(excursiedto.oraPlecare);
            String firma = excursiedto.firmaTransport;
            Double pret = excursiedto.pret;
            int locuri =excursiedto.locuriDisponibile;

            Excursie excursie = new Excursie(obiectiv,ora,firma,pret,locuri);
            excursie.setId(id);
            return excursie;
        }
        public static ExcursieDTO[] getDTO(Excursie[] excursii)
        {
            ExcursieDTO[] excursiiDto = new ExcursieDTO[excursii.Length];
            for (int i = 0;i< excursiiDto.Length;i++)
            {
                excursiiDto[i] = getDTO(excursii[i]);
            }
            return excursiiDto;
        }
        public static Excursie[] fromDTO(ExcursieDTO[] excursiidto)
        {
            Excursie[] excursii = new Excursie[excursiidto.Length];
            for(int i = 0; i< excursiidto.Length; i++)
            {
                excursii[i] = fromDTO(excursiidto[i]);
            }
            return excursii;
        }
    }
}
